xdsl/dialects/builtin.py

```diff
@@ -353,8 +353,39 @@ def print_parameter(self, printer: Printer) -> None:
                 raise ValueError(f"Invalid signedness {data}")
 
 
+class BitWidthType(TypeAttribute, ABC):
+    """
+    A type attribute with a defined bitwidth
+    """
+
+    name = "abstract.bitwidth_type"
+
+    @property
+    @abstractmethod
+    def get_bitwidth(self) -> int:
+        """
+        Returns the width of an element type in bits.
+        """
+        raise NotImplementedError()
```

Comment:
```suggestion
    @property
    @abstractmethod
    def bitwidth(self) -> int:
        """
        Returns the width of an element type in bits.
        """
        raise NotImplementedError()
```
If it's a property it might as well not have the get prefix